orangecontrib/prototypes/widgets/owface.py

```diff
@@ -33,8 +36,30 @@ def __init__(self):
                         checkbox_label="Run after any change",
                         orientation="horizontal")
 
+    def get_ext(self, file_path):
+        """Find the extension of a file or url."""
+        if not os.path.isfile(file_path):
+            file_path = urllib.parse.urlparse(file_path).path
+        return os.path.splitext(file_path)[1].strip().lower()
+
+    def read_img(self, file_path):
+        """Read an image from file or url and convert it to grayscale."""
+        try:
+            if os.path.isfile(file_path):
+                img = cv2.imread(file_path, cv2.IMREAD_GRAYSCALE)
+            else:
+                res = urllib.request.urlopen(file_path)
+                arr = np.asarray(bytearray(res.read()), dtype=np.uint8)

```

Comment:
Just wrap these two lines into try-except if this is where the error is to be expected.